src/record_layout.zig

```diff
@@ -0,0 +1,459 @@
+            var type_layout = TypeLayout.init(0, 0);
+
+            computeLayout(fld.ty, comp, &type_layout);
+
+            var field_attrs: ?[]const Attribute = null;
+            if (rec.field_attributes) |attrs| {
+                field_attrs = attrs[fld_indx];
+            }
+
+            if (fld.bit_width != null) {
+                self.layoutBitField(comp, fld, field_attrs, type_layout);
+            } else {
+                self.layoutRegluarField(fld, field_attrs, type_layout);
+            }
+        }
+    }
+    fn layoutRegluarField(
+        self: *SysVContext,
+        fld: *Field,
+        fld_attrs: ?[]const Attribute,
+        fld_layout: TypeLayout,
+    ) void {
+        var fld_align_bits = fld_layout.field_alignment_bits;
+
+        // If the struct or the field is packed, then the alignment of the underlying type is
+        // ignored. See test case 0084.
+        if (self.attr_packed or isPacked(fld_attrs)) {
+            fld_align_bits = BITS_PER_BYTE;
+        }
+
+        // The field alignment can be increased by __attribute__((aligned)) annotations on the
+        // field. See test case 0085.
+        if (annotationAlignment(fld_attrs)) |anno| {
+            fld_align_bits = std.math.max(fld_align_bits, anno);
+        }
+
+        // #pragma pack takes precedence over all other attributes. See test cases 0084 and
+        // 0085.
+        if (self.max_field_align_bits) |req_bits| {
+            fld_align_bits = std.math.min(fld_align_bits, req_bits);
+        }
+
+        // A struct field starts at the next offset in the struct that is properly
+        // aligned with respect to the start of the struct.
+        var offset_bits = if (self.is_union) 0 else std.mem.alignForwardGeneric(u64, self.size_bits, fld_align_bits);
+        var size_bits = fld_layout.size_bits;
+
+        // The alignment of a record is the maximum of its field alignments. See test cases
+        // 0084, 0085, 0086.
+        self.size_bits = std.math.max(self.size_bits, offset_bits + size_bits);
+        self.aligned_bits = std.math.max(self.aligned_bits, fld_align_bits);
+
+        fld.layout.offset_bits = offset_bits;
+        fld.layout.size_bits = size_bits;
+    }
+
+    fn layoutBitField(
+        self: *SysVContext,
+        comp: *const Compilation,
+        fld: *Field,
+        fld_attrs: ?[]const Attribute,
+        fld_layout: TypeLayout,
+    ) void {
+        const bit_width = fld.bit_width.?;
+
+        const ty_size_bits = fld_layout.size_bits;
+        var ty_fld_algn_bits: u32 = fld_layout.field_alignment_bits;
+
+        // Some targets ignore the alignment of the underlying type when laying out
+        // non-zero-sized bit-fields. See test case 0072. On such targets, bit-fields never
+        // cross a storage boundary. See test case 0081.
+        if (comp.ignoreNonZeroSizedBitfieldTypeAlignment()) {
+            ty_fld_algn_bits = 1;
+        }
+
+        if (bit_width <= 0) {
```

Comment:
```suggestion
        if (bit_width == 0) {
```
`bit_width` is unsigned so using less than or equal is misleading.

Reply:
old habit c of always using ranges when possible.